### PR TITLE
Reapply "Some cleanups enabled by removing support for Dart 1."

### PR DESCRIPTION
**Pull request by rmacnak-google**
- Add missing Dart 2 flags to gen_snapshot invocation.
- Disable brittle service test.

### CODE REVIEW

File: lib/snapshot/BUILD.gn [hunk 1]
```diff
@@ -25,203 +25,146 @@ copy("generate_dart_ui") {
...
+      vm_snapshot_instructions,
+      isolate_snapshot_data,
+      isolate_snapshot_instructions,
+    ]
 
-  rebased_dart_ui_path = rebase_path(dart_ui_path)
+    args = [
+      "--strong",
+      "--sync-async",
+      "--reify-generic-functions",
```
Comment (rmacnak-google) [inline lib/snapshot/BUILD.gn]: New part here.

File: shell/testing/observatory/test.dart [hunk 2]
```diff
@@ -156,7 +156,8 @@ final List<TestFunction> basicTests = <TestFunction>[
 ];
 
 final List<TestFunction> startPausedTests = <TestFunction>[
-  testStartPaused,
+  // TODO(engine): Investigate difference in lifecycle events.
```
Comment (rmacnak-google) [inline shell/testing/observatory/test.dart]: And here.

**Comment by cbracken** [comment]
lgtm

**Comment by cbracken** [comment]
This re-hammers-in the final nail in the coffin of #17051.

**Comment by goderbauer** [comment]
This seems to break coverage: https://build.chromium.org/p/client.flutter/builders/Linux/builds/6716/steps/run%20test.dart%20for%20coverage%20shard/logs/stdio

I'll revert it shortly.

**Pull request merged**